classes/webservice.php

```diff
@@ -268,6 +268,14 @@ private function make_call($path, $data = [], $method = 'get') {
         }
         $rawresponse = $this->make_curl_call($curl, $method, $url, $data);
 
+        $eofretries = 1;
+        while ($curl->get_errno() == 35 && $eofretries <= self::MAX_RETRIES) {
+            sleep(1);
+            debugging('retrying after curl error 35, retry attempt ' . $eofretries);
+            $eofretries++;
+            $rawresponse = $this->make_curl_call($curl, $method, $url, $data);
+        }
```

Comment:
I like how isolated the current version is, but I typically like to fold the base case into the process. If you like this suggestion, great! But it's not something that I'm going to require it, so please let me know your thoughts.
```suggestion

        $attempts = 0;
        do {
            if ($attempts > 0) {
                sleep(1);
                debugging('retrying after curl error 35, retry attempt ' . $attempts);
            }

            $rawresponse = $this->make_curl_call($curl, $method, $url, $data);
            $attempts++;
        } while ($curl->get_errno() === 35 && $attempts <= self::MAX_RETRIES);
```

Reply:
I'm fine with the change, only possible issue is maintenance since it requires a little more skill to read.